gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/oauth/OAuthViewModel.kt

```diff
@@ -30,19 +32,19 @@ internal class OAuthViewModel(
         }
     }
 
-    fun fetchAccessToken(code: String, clientId: String, clientSecret: String, redirectUri: String) {
+    fun fetchAccessToken(code: String, oAuthParams: OAuthParams, email: String) {
         viewModelScope.launch {
             _uiState.update { currentState -> currentState.copy(isAuthorizing = true) }
             val result = wordPressOAuthService.getAccessToken(
                 code = code,
-                clientId = clientId,
-                clientSecret = clientSecret,
-                redirectUri = redirectUri,
+                clientId = oAuthParams.clientId,
+                clientSecret = oAuthParams.clientSecret,
+                redirectUri = oAuthParams.redirectUri,
             )
 
             when (result) {
                 is Result.Success -> {
-                    // todo: Save access token
+                    tokenStorage.storeToken(email, result.value)
```

Comment:
We can use the `hash` instead of the `email` as key in the `tokenStorage`. I guess I someone obtain the `token` we have a big problem but a least they won't have the relation between `mail<->token`.

Reply:
Good idea! Done